docs/reference/index.rst

```diff
@@ -0,0 +1,44 @@
+API Reference
+=============
+
+ImageIO's API follows the usual idea of choosing sensible defaults for the average use, but giving
+you fine grained control 
+
+Core API (Basic Usage)
+^^^^^^^^^^^^^^^^^^^^^^
+
+The API documented in this section is shared by all plugins
+
+.. toctree::
+    :maxdepth: 1
+
+    Core API v2.9 <userapi>
```

Comment:
It looks like the new API is not documented yet, correct?

Reply:
Correct. I wanted to use this PR to document the new API, but then realized that I made a lot of commits to refactor the docs to "get things read for documentation". Because of that I decided to make this PR about refactoring things, and the next PR about actually writing docs for it and updating the examples, etc.

This way we can have these docs for v2.9 and then have updated docs and examples for v2.10/v3.0.

I also decided to clean up some of the doc issues in the process, since I am shuffling around a lot of content and adding a paragraph or note here and there is easy to do.